src/lab3-tests.md

```diff
@@ -0,0 +1,115 @@
+Tests for WBE Chapter 3
+=======================
+
+Chapter 3 (Formatting Text) adds on font metrics and simple font styling via
+HTML tags. This file contains tests for the additional functionality.
+
+    >>> import test
+    >>> _ = test.socket.patch().start()
+    >>> _ = test.ssl.patch().start()
+    >>> import lab3
+
+Testing `lex`
+-------------
+
+The `lex` function in chapter three has been beefed up to return an array
+of `Tag` or `Text` objects, rather than just the stream of characters from the
+input.
+
+    >>> lab3.lex('<body>hello</body>')
+    [Tag: body, Text: hello, Tag: /body]
```

Comment:
I'm in favor of adding `__repr__` methods to our objects, for testing, but let's not have unquoted user strings here—it could get confusing. Something like `Tag('body')` would be a bit wordier but is less likely to confuse someone accidentally.

Reply:
Done